packages/mapviewer/src/api/layers/WMSCapabilitiesParser.class.js

```diff
@@ -431,8 +431,41 @@ export default class WMSCapabilitiesParser {
         return [new LayerAttribution(title, url)]
     }
 
+    _constructLegendFromGetLegendGraphic(getLegendGraphicObject, layerId) {
+        return (
+            getLegendGraphicObject.DCPType[0].HTTP.Get.OnlineResource +
+            `SERVICE=WMS` +
+            `&REQUEST=GetLegendGraphic` +
+            `&VERSION=${this.version}` +
+            `&FORMAT=${getLegendGraphicObject.Format[0]}` +
+            `&LAYER=${layerId}` +
+            `&SLD_VERSION=1.1.0`
+        )
+    }
+
     _getLayerLegends(layerId, layer) {
         const styles = layer.Style?.filter((s) => s.LegendURL?.length > 0) ?? []
+        // if we do not have access to the legend in pure WMS fashion, we check if this
+        // WMS follows the SLD specification, and if we can get it from there.
+        if (
+            styles.length === 0 &&
+            layer.queryable &&
+            this.Capability.UserDefinedSymbolization?.SupportSLD &&
+            this.Capability.Request.GetLegendGraphic?.DCPType[0]?.HTTP?.Get?.OnlineResource
+        ) {
+            const url = this._constructLegendFromGetLegendGraphic(
```

Comment:
in prep for TS migration, I'd prefer you set up a couple `const` here from the `this.Capability.Request.GetLegendGraphic` object. It will be way easier to type later on (and TS will want to have the guarding if to accept we deconstruct the `this.Capability.Request.GetLegendGraphic` object)
Then you can build the URL in place (no need for the `_constructLegendFromGetLegendGraphic` function, it's only used once here)